### ISSUE TITLE
Issue with zio-prelude 1.0.0-RC8-1 (zio1) macros and Scala 2.13.[9|10]

### ISSUE THREAD

**Issue by guizmaii** [zio/zio-prelude#969]
Hi everyone 🙂

Since Scala 2.13.9, our prelude's newtypes are not compiling anymore.
I reported the issue to Scala here: https://github.com/scala/bug/issues/12647

Lukas Rytz suggests making a change in prelude's macros to make them work with Scala 2.13.9+ here: https://github.com/scala/bug/issues/12647#issuecomment-1275760141

I don't have enough time to try to make this change for now that's why I'm reporting this issue.
I'll probably not have any time to try to make a PR before a week or two.

Cheers,
Jules

**Comment by sideeffffect** [comment]
@kitlangton you our macro guru. If you have the time, maybe you can have a look at this? :bow:

**Comment by som-snytt** [comment]
I'm new to this code, but I suggest on the Scala ticket that it might be enough to
```
-  def assertion: QuotedAssertion[A] = new QuotedAssertion[A] {
+  def assertion: Any = new QuotedAssertion[A] {
```
where the `Any` magically makes the compiler use the refined RHS.

I haven't tried more than a simple test under Scala 2 only.

**Comment by guizmaii** [comment]
FYI, @som-snytt made a PR on Scala to fix this issue: https://github.com/scala/scala/pull/10188

**Comment by guizmaii** [comment]
Fixed in https://github.com/scala/scala/pull/10188

Fix will be released with Scala 2.13.11
